solutions/usaco-1181.mdx

```diff
@@ -0,0 +1,97 @@
+---
+id: usaco-1181
+source: USACO Bronze 2022 January
+title: Drought
+author: Chongtian Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_prob3_bronze_jan22.html)
+
+## Explanation
+
+We can solve this problem greedily using one linear sweep and making prefixes of cows have equal hunger as we sweep from left to right. Thus, we can break this
+problem up into cases:
+
+Suppose the cows are numbered from $1$ to $n$, and we are currently looking at cow $i$. If cow $i+1$ has a greater hunger value, then it is sufficient to decrease cow $i+1$ to the same hunger valueas cow $i$. To do this we will pick the pair of cows $i+1$ and $i+2$ and decrease their hungers until cow $i+1$ has the same hunger value as cow $i$. Of course, if cow $i+2$ has a negative hunger value, then it is impossible and we output $-1$.
+
+Now suppose cow $i$ has a greater hunger value than cow $i+1$. We obviously cannot decrease the hunger for cow $i+1$ any further since it will never be
+equal - we only can decrease the hunger for all the cows before $i+1$. To do this we select adjacent pairs of cows starting with the odd numbered cows and
+decrease their hunger values together (cows $1$ and $2$, cows $3$ and $4$, $\ldots$, cows $i-1$ and $i$). After this, all the cows before $i+1$ will be decreased to the same hunger value as cow $i+1$. However, when $i$ is odd, it is impossible to decrease it's hunger unless we also decrease the hunger value for cow $i+1$, so we output $-1$.
```

Comment:
this giant wall of text is very intimidating- perhaps walk the reader through with an example

Reply:
what do you mean? am i supposed to walk them through a sample case?

Reply:
just like
```
5 5 5 3 2
```
let's say we were at the last 5
[explanation follows]

Reply:
is it necessary? i don't see other bronze explanations do it.

Reply:
does drawings of how the steps work suffice?

Reply:
actually yeah, i can do that with a drawing

Reply:
@SansPapyrus683 is this fine?
https://cdn.discordapp.com/attachments/932459332946264144/963998668921389066/Screen_Shot_2022-04-13_at_11.06.10_PM.png
If so, how do I embed images in mdx?

Reply:
i'd prefer you do it with some ascii stuff
for example
```
5 5 5 5 3 2
4 4 4 4 3 2
3 3 3 3 3 2
```

Reply:
ive done it, but i dont know how to do the fancy mdx ascii formatting. could you fix it?